### PR TITLE
Support special nuclei reconstruction

### PR DESCRIPTION
**Pull request by mpuccio**
This is to support the special PID for tracking for nuclei being introduced with alisw/AliRoot#1017
Needs to be tested before merging!

cc. @akalweit

### CODE REVIEW

**Comment by miweberSMI** [comment]
Hi @mpuccio, @akalweit, all,

what is the status of this PR?

Thanks and cheers,
Michael

**Comment by ktf** [comment]
any updates?

**Comment by mpuccio** [comment]
Ready to be merged for me!

File: MC/ReconstructionConfig.C [hunk 1]
```diff
@@ -203,6 +205,16 @@ void ReconstructionConfig(AliReconstruction &rec, int tag_tmp)
     rec.SetRun1PIDforTracking(kTRUE);
     return;
 
+  case kReconstructionForNuclei:
+    ReconstructionDefault(rec);
+    const char* conf = gSystem->Getenv("CONFIG_HE3_PION_THRESHOLD");
+    int threshold = 130;
+    if (conf && !conf[0]) {
```
Comment (fprino) [inline MC/ReconstructionConfig.C:212]: Hi Max, why did you put the condition !conf[0] ?

File: DataProc/PPass/main_rec.C [hunk 2]
```diff
@@ -129,6 +129,13 @@ void main_rec(const char *filename="raw.root", const char* options="")
   rec.SetDeleteRecPoints(delRecPoints.Data()); 
   //
 
+  const char* confPID = gSystem->Getenv("CONFIG_HE3_PION_THRESHOLD");
```
Comment (fprino) [inline DataProc/PPass/main_rec.C:132]: Hi,
if I am not wrong, the environment variable that are set in the JDL (and I guess we will need to do it via JDL also for this special reco) have a prefix ALICE_JDL that is added automatically.
@catalinristea can confirm.
If this is the case, you should use ALICE_JDL_CONFIG_HE3_PION_THRESHOLD here.

File: MC/ReconstructionConfig.C (hunk 1 above)
Comment (mpuccio) [reply MC/ReconstructionConfig.C]: That is the old piece of code, I now replaced it with the one that uses only atoi, thanks for noticing!

File: DataProc/PPass/main_rec.C (hunk 2 above)
Comment (mpuccio) [reply DataProc/PPass/main_rec.C]: I gather from the other variables around that the prefix is `ALIEN_JDL_`, I adapted the new variable to this naming convention. But let's see what @catalinristea says! Thanks for noticing also this.

Comment (chiarazampolli) [reply DataProc/PPass/main_rec.C]: Yes, I also confirm that you need the prefix if you want to read an env variable set in the JDL.

**Comment by catalinristea** [comment]
Ciao,

I'm testing in grid to make sure it works. I'm not sure about the underscores in the name.
Thanks.

Yours,
Catalin.

File: DataProc/PPass/main_rec.C (hunk 2 above)
Comment (catalinristea) [reply DataProc/PPass/main_rec.C]: Ciao,

I'm testing in grid to make sure it works. I'm not sure about the underscores in the name.
Thanks.

Yours,
Catalin.

Comment (chiarazampolli) [reply DataProc/PPass/main_rec.C]: Thanks Catalin. I guess you have doubts in the underscores in the "CONFIG_HE3_PION_THRESHOLD" part, right? There I don't know if they are accepted or not. We can check with @costing ?

Comment (catalinristea) [reply DataProc/PPass/main_rec.C]: Ciao,

I confirm it works. Please go on and merge, if nothing else.
Thanks!

Yours,
Catalin.

Comment (chiarazampolli) [reply DataProc/PPass/main_rec.C]: Done, thanks!

**Pull request merged**